continuous-integration/readme.md

```diff
@@ -1,3 +1,235 @@
+  - [Developer Access to Latest Release Artifacts](#developer-access-to-latest-release-artifacts)
+  - [Integration Observability](#integration-observability)
+  - [Resources](#resources)
+
+## Goals
+
+Continuous integration automation is an integral part of the software development lifecycle intended to reduce build integration errors and maximize velocity across a dev crew.
+
+A robust build automation pipeline will:
+
+- Accelerate team velocity
+- Prevent integration problems
+- Avoid last minute chaos during release dates
+- Provide a quick feedback cycle for system-wide impact of local changes
+- Separate build and deployment stages
+- Measure and report metrics around build failures / success(s)
+- Increase visibility across the team enabling tighter communication
+- Reduce human errors, which is probably the most important part of automating the builds
+
+## S/ingle Source Repository
+
+- [ ] **Code / manifest artifacts required to build your project should be maintained in a single git repository.**
+
+  - CI provider-specific build [pipeline definition(s)](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops#define-pipelines-using-yaml-syntax) should reside within the mainline repository.
+
+## Build Automation
+
+An automated build should encompass the following principles:
+
+- [ ] **Build Task**
+  - A single step within your build pipeline that compiles your code project into a single build artifact.
+- [ ] **Unit Testing**
+  - Your build definition includes validation steps to execute a suite of automated unit tests to ensure that application components meets its design and behaves as intended.
+- [ ] **Code Style Checks**
+  - Code across an engineering team must be formatted to agreed coding standards. Such standards keep code consistent and most importantly easy for the team and customer(s) to read and refactor. Code styling consistency encourages collective ownership for project scrum teams and our partners.
+  - There are several open source code style validation tools available to choose from ([code style checks](https://github.com/checkstyle/checkstyle), [StyleCop](https://en.wikipedia.org/wiki/StyleCop)). The [Code Review section](https://github.com/microsoft/code-with-engineering-playbook/tree/master/code-reviews#language-specific-guidance) of the playbook has suggestions for linters and preferred styles for a number of languages.
+  - Code standards are maintained within a single configuration file. There should be a step in your build pipeline that asserts code in the latest commit conforms to the known style definition.
+- [ ] **Database Schema Automation**
+  - The build process should automate pulling database schemas from the git repository and applied to the target environment. This also includes automating the steps to seed that environment with the necessary test data required to run isolated end-to-end automated tests.
```

Comment:
the build process should not do this, nor should the release process. The tests themselves should be responsible for seeding & cleaning up test data

Reply:
I think it depends.... I would agree with your statement if the test dataset was small... However, if the test dataset size is north of a Gigibyte or even a Terrabyte then a team would consider bulk copying the data to the target environment. 

You bring up a good point where the scenario(s) can vary and the guidance for staging test data in an automated way may not belong as a CI principle. 

That being said, I'm going to strike this bullet for now.